enarx-keep-sev-shim/src/usermode.s

```diff
@@ -0,0 +1,45 @@
+# SPDX-License-Identifier: Apache-2.0
+
+.section .text, "ax"
+.global usermode
+.type usermode, @function
+.code64
+
+# %rdi: instruction pointer
+# %rsi: stack pointer
+# %rdx: argument
+
+.p2align 4
+usermode:
+    pushq   $0x1b          # USER_DATA_SEGMENT
+    pushq   %rsi           # stack pointer
+# FIXME: disable interrupts for now, until we need them
+#   movq    $0x200,%r12    # RFlags (1 << 9) - set interrupt enable flag
+    pushq   $0x000         # RFlags
+    pushq   $0x23          # USER_CODE_SEGMENT
```

Comment:
I can't use assembler statics with .equ because it's an absolute, which conflicts with pie.
This will go away, once we have stable asm! and #[naked_function]